File: sql-insight-engine/sql-insight-engine-innodb/src/main/java/org/gongxuanzhang/sql/insight/core/engine/storage/innodb/index/SecondIndex.java
package org.gongxuanzhang.sql.insight.core.engine.storage.innodb.index;


import org.gongxuanzhang.sql.insight.core.environment.SessionContext;
import org.gongxuanzhang.sql.insight.core.object.Column;
import org.gongxuanzhang.sql.insight.core.object.Cursor;
import org.gongxuanzhang.sql.insight.core.object.InsertRow;
import org.gongxuanzhang.sql.insight.core.object.Table;

import java.io.File;
import java.util.List;

/**
 * second index
 *
 * @author gongxuanzhangmelt@gmail.com
 **/
public class SecondIndex extends InnodbIndex {


    protected SecondIndex(Table table) {
        super(table);
    }

    @Override
    public void rndInit() {

    }

    @Override
    public int getId() {
        return 0;
    }

    @Override
    public Table belongTo() {
        return null;
    }

    @Override
    public Cursor find(SessionContext sessionContext) {
        return null;
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public void insert(InsertRow row) {

    }

    @Override
    public File getFile() {
        return null;
    }

    @Override
    public List<Column> columns() {
        return null;
    }

}

File: sql-insight-engine/sql-insight-engine-innodb/src/main/kotlin/tech/insight/engine/innodb/index/SecondIndex.kt
//package tech.insight.engine.innodb.index
//
//import tech.insight.core.bean.Column
//import tech.insight.core.bean.Cursor
//import tech.insight.core.bean.InsertRow
//import tech.insight.core.bean.Table
//import tech.insight.core.environment.Session
//import tech.insight.engine.innodb.page.IndexKey
//import tech.insight.engine.innodb.page.InnoDbPage
//import java.io.File
//
///**
// * second index
// *
// * @author gongxuanzhangmelt@gmail.com
// */
//class SecondIndex protected constructor(table: Table) : InnodbIndex() {
//    override fun insert(row: InsertRow) {
//        TODO("Not yet implemented")
//    }
//
//    override fun findByKey(key: IndexKey): InnoDbPage? {
//        TODO("Not yet implemented")
//    }
//
//    override fun rndInit() {
//        TODO("Not yet implemented")
//    }
//
//    override val id: Int
//        get() = TODO("Not yet implemented")
//
//    override fun find(session: Session): Cursor {
//        TODO("Not yet implemented")
//    }
//
//    override val name: String
//        get() = TODO("Not yet implemented")
//    override val file: File
//        get() = TODO("Not yet implemented")
//
//    override fun columns(): List<Column> {
//        TODO("Not yet implemented")
//    }
//}
